pyfftw/interfaces/_utils.py

```diff
@@ -97,24 +101,25 @@ def _Xfftn(a, s, axes, overwrite_input, planner_effort,
         # take the transform of the input array! (in general, we have
         # to assume that the input array will be destroyed during 
         # planning).
-        a_copy = a.copy()
+        if overwrite_input:    # Note: reload_after_transform is not True...
+            args = (a, s, axes) + args
+            a_copy = a.copy()  # ... hence this is not an additional copy
+        else:
+            args = (numpy.empty_like(a), s, axes) + args
```

Comment:
This is the main potential issue and does not seem to be covered in the unit tests.  Can you give me an example?  I tried the following, but it seems to work fine.

```python
import numpy as np
from pyfftw.interfaces import numpy_fft as npfft
A = np.random.random((10,10))
B = A[0::2, 0::2]  # Non-contiguous input array.
np.allclose(np.fft.fftn(B), npfft.fftn(B))
```